test/local/mysql_tests.js

```diff
@@ -1,388 +1,369 @@
-var zeroBuffer32 = Buffer.from('0000000000000000000000000000000000000000000000000000000000000000', 'hex')
-var now = Date.now()
+const assert = require('insist')
+const dbServer = require('../../fxa-auth-db-server')
+const log = require('../lib/log')
+const DB = require('../../lib/db/mysql')(log, dbServer.errors)
+const config = require('../../config')
+const P = require('../../lib/promise')
+const crypto = require('crypto')
 
-DB.connect(config)
-  .then(
-    function (db) {
+const zeroBuffer16 = Buffer.from('00000000000000000000000000000000', 'hex')
+const zeroBuffer32 = Buffer.from('0000000000000000000000000000000000000000000000000000000000000000', 'hex')
+const now = Date.now()
 
-      test(
-        'ping',
-        function (t) {
-          t.plan(1)
-          return db.ping()
-          .then(function(account) {
-            t.pass('Got the ping ok')
-          }, function(err) {
-            t.fail('Should not have arrived here')
-          })
-        }
-      )
-
-      test(
-        'a select on an unknown table should result in an error',
-        function (t) {
-          var query = 'SELECT mumble as id FROM mumble.mumble WHERE mumble = ?'
-          var param = 'mumble'
-          db.read(query, param)
-            .then(
-              function(result) {
-                t.plan(1)
-                t.fail('Should not have arrived here for an invalid select')
-              },
-              function(err) {
-                t.plan(5)
-                t.ok(err, 'we have an error')
-                t.equal(err.code, 500)
-                t.equal(err.errno, 1146)
-                t.equal(err.error, 'Internal Server Error')
-                t.equal(err.message, 'ER_NO_SUCH_TABLE')
-              }
-            )
-        }
-      )
-
-      test(
-        'an update to an unknown table should result in an error',
-        function (t) {
-          var query = 'UPDATE mumble.mumble SET mumble = ?'
-          var param = 'mumble'
+describe('MySQL', () => {
 
-          db.write(query, param)
-            .then(
-              function(result) {
-                t.plan(1)
-                t.fail('Should not have arrived here for an invalid update')
-              },
-              function(err) {
-                t.plan(5)
-                t.ok(err, 'we have an error')
-                t.equal(err.code, 500)
-                t.equal(err.errno, 1146)
-                t.equal(err.error, 'Internal Server Error')
-                t.equal(err.message, 'ER_NO_SUCH_TABLE')
-              }
-            )
-        }
-      )
+  let db
+  before(() => {
+    return DB.connect(config).then(db_ => {
+      db = db_
+    })
+  })
 
-      test(
-        'an transaction to update an unknown table should result in an error',
-        function (t) {
-          var sql = 'UPDATE mumble.mumble SET mumble = ?'
-          var param = 'mumble'
+  it(
+    'ping',
+    () => {
+      return db.ping()
```

Comment:
In one of the other modules you just pulled `db.ping()` up into `before`, I quite liked that fwiw.